Source/ASCollectionView.mm

```diff
@@ -970,45 +979,118 @@ - (CGSize)collectionView:(UICollectionView *)collectionView layout:(UICollection
   return [self sizeForElement:element];
 }
 
-- (CGSize)collectionView:(UICollectionView *)collectionView layout:(UICollectionViewLayout *)layout referenceSizeForHeaderInSection:(NSInteger)section
+- (NSIndexPath *)pendingIndexPathForSection:(NSInteger)section
+{
+  // NOTE: For now, we don't translate length = 1 NSIndexPaths between spaces. Use the 0th item until then.
+  NSIndexPath *sectionIndexPath = [NSIndexPath indexPathForItem:0 inSection:section];
+  ASCollectionElement *element = [_dataController.visibleMap elementForItemAtIndexPath:sectionIndexPath];
+  NSIndexPath *pendingIndexPath = element ? [_dataController.pendingMap indexPathForElement:element] : nil;
+  return pendingIndexPath;
```

Comment:
Should this be `- (NSInteger)dataSourceSectionForSection:(NSInteger)section`? I could see not-checking-NSNotFound on the caller side being a problem. Not checking nil is roughly as likely to occur but slightly less likely to cause issues immediately.

Currently `ASSection` objects are only created if the user implements the optional `contextForSection:` API. We should instead create `ASSections` unconditionally and use them as identifiers, similar to ASCollectionElement. That way looking up the indexes of sections between maps is easy!!